azkaban-common/src/main/java/azkaban/executor/ExecutionOptions.java

```diff
@@ -166,6 +175,52 @@ public static ExecutionOptions createFromObject(final Object obj) {
     return options;
   }
 
+  /**
+   * Validate the ExecutionOption's Flow-Parameters against the application level Properties
+   * @throws ServletException if any of the parameter is invalid
+   */
+  public void validateFlowParameters(Props azProps)
+      throws ServletException {
+    List<String> errMsg = new ArrayList<>();
+
+    Map<String, String> flowParameters = this.getFlowParameters();
+    if (flowParameters.containsKey(FlowParameters.FLOW_PARAM_ALLOW_RESTART_ON_STATUS)) {
+      // allow list defined in azProps
+      final List<String> allowedStatuses = azProps.getStringList(
+          AZKABAN_EXECUTION_RESTARTABLE_STATUS, DEFAULT_EXECUTION_RESTARTABLE_STATUS);
+
+      // user defined list
+      final String[] statuses = flowParameters
+          .getOrDefault(FlowParameters.FLOW_PARAM_ALLOW_RESTART_ON_STATUS, "")
+          .split("\\s*,\\s*");
+
+      for (String s : statuses) {
+        if (!allowedStatuses.contains(s)) {
+          errMsg.add(String.format("`%s` is not a valid restartable status, "
+              + "permitted status are %s", s, allowedStatuses));
+        }
+      }
+    }
+    if (flowParameters.containsKey(FlowParameters.FLOW_PARAM_RESTART_COUNT)){
+      // check restart count limit
+      final int azFlowRestartCountLimit = azProps.getInt(
```

Comment:
I think adding the prefix az to this variable name is redundant. Please consider removing this.

Reply:
I am good either way, removing